libs/language-server/src/lib/ast/model-util.ts

```diff
@@ -20,101 +15,14 @@ import {
   PipelineDefinition,
   UnaryExpression,
   isBuiltinBlocktypeDefinition,
-  isCompositeBlocktypeDefinition,
   isJayveeModel,
 } from './generated/ast';
 // eslint-disable-next-line import/no-cycle
-import { BlockTypeWrapper, ConstraintTypeWrapper } from './wrappers';
 import {
-  PipeWrapper,
-  createWrappersFromPipeChain,
-} from './wrappers/pipe-wrapper';
-
-export function collectStartingBlocks(
-  container: PipelineDefinition | CompositeBlocktypeDefinition,
-): BlockDefinition[] {
-  // For composite blocks the first blocks of all pipelines are starting blocks as they have inputs
-  if (isCompositeBlocktypeDefinition(container)) {
-    const startingBlocks = container.pipes
-      .map((pipe) => pipe.blocks[0])
-      .map((blockRef: Reference<BlockDefinition> | undefined) => {
-        if (
-          blockRef?.ref !== undefined &&
-          BlockTypeWrapper.canBeWrapped(blockRef.ref.type)
-        ) {
-          return blockRef.ref;
-        }
-        return undefined;
-      })
-      .filter((x): x is BlockDefinition => x !== undefined);
-
-    return startingBlocks;
-  }
-
-  const result: BlockDefinition[] = [];
-
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-  const blocks = container?.blocks ?? [];
-  for (const block of blocks) {
-    if (!BlockTypeWrapper.canBeWrapped(block.type)) {
-      continue;
-    }
-    const blockType = new BlockTypeWrapper(block.type);
-
-    if (!blockType.hasInput()) {
-      result.push(block);
-    }
-  }
-  return result;
-}
-
-export function collectChildren(block: BlockDefinition): BlockDefinition[] {
```

Comment:
I like that we got rid of these util functions and moved to the wrapper concept we are using in other places already, well done :).

Reply:
Should I also relocate the function `getBlocksInTopologicalOrder`?

Reply:
I think that makes sense as well, yeah.

Reply:
Addressed in [2421058](https://github.com/jvalue/jayvee/pull/491/commits/2421058f33ee8ae19b802984baeed38a4d5ec244)